src/jdk.crypto.ec/share/classes/sun/security/ec/ECPrivateKeyImpl.java

```diff
@@ -244,26 +244,29 @@ long getNativePtr() {
                     byte[] gy = generator.getAffineY().toByteArray();
                     byte[] n = this.params.getOrder().toByteArray();
                     byte[] h = BigInteger.valueOf(this.params.getCofactor()).toByteArray();
-                    byte[] p = new byte[0];
+                    long nativePointer;
                     if (field instanceof ECFieldFp) {
+                        byte[] p = new byte[0];
                         p = ((ECFieldFp)field).getP().toByteArray();
```

Comment:
The main motivation for moving the declaration was to avoid the allocation (`new byte[0]`); we want:
```
                        byte[] p = ((ECFieldFp)field).getP().toByteArray();
```